packages/core/src/entity/Collection.ts

```diff
@@ -299,9 +299,13 @@ export class Collection<T extends object, O extends object = object> extends Arr
 
     if (options.dataloader ?? [DataloaderType.ALL, DataloaderType.COLLECTION].includes(DataloaderUtils.getDataloaderType(em.config.get('dataloader')))) {
       const order = [...this.items]; // copy order of references
-      const customOrder = !!options.orderBy;
+      const orderBy = this.createOrderBy(options.orderBy);
+      const customOrder = !!orderBy;
```

Comment:
this method always returns an array, so you cant just do `!!orderBy`, it will always be true

```suggestion
      const orderBy = this.createOrderBy(options.orderBy);
      const customOrder = orderBy.length > 0;
```

(uncovered by the coverage report)